tests/e2e/actor_reminder_partition/actor_reminder_partition_test.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/dapr/dapr/tests/runner"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	appName                      = "actorreminderpartition"                      // App name in Dapr.
+	actorIDPartitionTemplate     = "actor-reminder-partition-test-%d"            // Template for Actor ID.
+	reminderName                 = "PartitionTestReminder"                       // Reminder name.
+	numIterations                = 7                                             // Number of times each test should run.
+	numHealthChecks              = 60                                            // Number of get calls before starting tests.
+	numActorsPerThread           = 10                                            // Number of get calls before starting tests.
+	secondsToCheckReminderResult = 20                                            // How much time to wait to make sure the result is in logs.
+	actorInvokeURLFormat         = "%s/test/testactorreminderpartition/%s/%s/%s" // URL to invoke a Dapr's actor method in test app.
+	actorlogsURLFormat           = "%s/test/logs"                                // URL to fetch logs from test app.
+)
+
+// represents a response for the APIs in this app.
+type actorLogEntry struct {
+	Action         string `json:"action,omitempty"`
+	ActorType      string `json:"actorType,omitempty"`
+	ActorID        string `json:"actorId,omitempty"`
+	StartTimestamp int    `json:"startTimestamp,omitempty"`
+	EndTimestamp   int    `json:"endTimestamp,omitempty"`
+}
+
+type actorReminder struct {
+	Data     string `json:"data,omitempty"`
+	DueTime  string `json:"dueTime,omitempty"`
+	Period   string `json:"period,omitempty"`
+	Callback string `json:"callback,omitempty"`
+}
+
+func parseLogEntries(resp []byte) []actorLogEntry {
+	logEntries := []actorLogEntry{}
+	err := json.Unmarshal(resp, &logEntries)
+	if err != nil {
+		return nil
+	}
+
+	return logEntries
+}
+
+func countActorAction(resp []byte, actorID string, action string) int {
+	count := 0
+	logEntries := parseLogEntries(resp)
+	for _, logEntry := range logEntries {
+		if (logEntry.ActorID == actorID) && (logEntry.Action == action) {
+			count++
+		}
+	}
+
+	return count
+}
+
+var tr *runner.TestRunner
+
+func TestMain(m *testing.M) {
+	// These apps will be deployed before starting actual test
+	// and will be cleaned up after all tests are finished automatically
+	testApps := []kube.AppDescription{
+		{
+			AppName:        appName,
+			DaprEnabled:    true,
+			ImageName:      "e2e-actorfeatures",
+			Replicas:       1,
+			IngressEnabled: true,
+			MetricsEnabled: true,
+			DaprCPULimit:   "2.0",
+			DaprCPURequest: "0.1",
+			AppCPULimit:    "2.0",
+			AppCPURequest:  "0.1",
+			AppEnv: map[string]string{
+				"TEST_APP_ACTOR_TYPE": "testactorreminderpartition",
+			},
+			Config: "actortypemetadata",
+		},
+	}
+
+	tr = runner.NewTestRunner(appName, testApps, nil, nil)
+	os.Exit(tr.Start(m))
+}
+
+func TestActorReminder(t *testing.T) {
```

Comment:
Can we expand the migration tests? Right now this is just 0 -> 5, right? It'd be good to see that an existing partition set can be migrated and that the decrement case does what we expect.

Reply:
I can add a scenario to add more partitions. The decrement does not work by design and I will explain why in the comment below.

Reply:
I've modified the existing reminder partition e2e test to also increment number of partitions midway.